app/models/manageiq/providers/embedded_terraform/automation_manager/job.rb

```diff
@@ -0,0 +1,129 @@
+class ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Job < Job
+  def self.create_job(template, env_vars, input_vars, credentials, poll_interval: 1.minute)
+    super(
+      :template_id   => template.id,
+      :env_vars      => env_vars,
+      :input_vars    => input_vars,
+      :credentials   => credentials,
+      :poll_interval => poll_interval,
+    )
+  end
+
+  def start
+    queue_signal(:pre_execute)
+  end
+
+  def pre_execute
+    checkout_git_repository
+    signal(:execute)
+  end
+
+  def execute
+    template_path = File.join(options[:git_checkout_tempdir], template_relative_path)
+
+    response = Terraform::Runner.run(
+      options[:input_vars],
+      template_path,
+      :credentials => options[:credentials],
+      :env_vars    => options[:env_vars]
+    )
+
+    options[:terraform_stack_id] = response.stack_id
+    save!
+
+    queue_poll_runner
+  end
+
+  def poll_runner
+    if running?
+      queue_poll_runner
+    else
+      signal(:post_execute)
+    end
+  end
+
+  def post_execute
+    cleanup_git_repository
+
+    success? ? queue_signal(:finish, message, status) : abort_job("Failed to run template", "error")
+  end
+
+  alias initializing dispatch_start
+  alias finish       process_finished
+  alias abort_job    process_abort
+  alias cancel       process_cancel
+  alias error        process_error
+
+  protected
+
+  def running?
+    stack_response&.running?
+  end
+
+  def success?
+    stack_response&.response&.status == "SUCCESS"
+  end
+
+  def load_transitions
+    self.state ||= 'initialize'
+
+    {
+      :initializing => {'initialize'       => 'waiting_to_start'},
+      :start        => {'waiting_to_start' => 'pre_execute'},
+      :pre_execute  => {'pre_execute'      => 'execute'},
+      :execute      => {'execute'          => 'running'},
+      :poll_runner  => {'running'          => 'running'},
+      :post_execute => {'running'          => 'post_execute'},
+      :finish       => {'*'                => 'finished'},
+      :abort_job    => {'*'                => 'aborting'},
+      :cancel       => {'*'                => 'canceling'},
+      :error        => {'*'                => '*'}
+    }
+  end
+
+  def poll_interval
+    options.fetch(:poll_interval, 1.minute).to_i
```

Comment:
1 minute seems like a long time.  How long are we expecting short and long templates to take to run?  Would 30 seconds be better?

Reply:
Probably a good question for @putmanoj